interfaces/apparmor/backend.go

```diff
@@ -166,7 +166,7 @@ func (b *Backend) Initialize() error {
 	}
 
 	// We are not using apparmor.LoadProfile() because it uses other cache.
-	if err := loadProfile(profilePath, dirs.SystemApparmorCacheDir); err != nil {
+	if err := loadProfile(profilePath, dirs.SystemApparmorCacheDir, skipReadCache); err != nil {
```

Comment:
This will result in a small performance hit since we will unconditionally not ever use the cache with the snap-confine profile. This is probably fine for this PR until the apparmor hash can be reliably used. Perhaps add a TODO comment to remove once that is available?

Reply:
We do check if the profile has changed above, if we didn't change it we're not even reaching this place. It was an optimization of a specialized statement already. As such I don't believe it is an issue.

Reply:
Maybe I'm blind, but AFAICT you are only checking if /var/lib/snapd/apparmor/snap-confine changes. Again, this is not a blocker, just something to call out.

Reply:
Let me look at this again just to be sure.

Reply:
If you look a few lines above this part, in https://github.com/zyga/snapd/blob/fix/apparmor-mtime-resiliency/interfaces/apparmor/backend.go#L133 you will see that apparmor.Backend.Initialize returns if there is nothing created or removed (no snap-confine profile changes). This means that we never reach the code you referenced.

When things do change, since this Initialize runs early during the startup of snapd, when be may still not be NTP-synchronized, I specifically want to recompile the profile ignoring the cache to ensure we don't have this issue in another forum re-surfacing again.

Reply:
The line you mention is only after looking at dirs.SnapConfineAppArmorDir, ie /var/lib/snapd/apparmor/snap-confine. Again, I don't see this as a problem.